onnxruntime/core/session/ort_env.cc

```diff
@@ -20,6 +20,8 @@ std::unique_ptr<OrtEnv> OrtEnv::p_instance_;
 int OrtEnv::ref_count_ = 0;
 onnxruntime::OrtMutex OrtEnv::m_;
 
+std::atomic_bool OrtEnv::dtor_has_run_once_{false};
```

Comment:
The approach seems reasonable. Do we have any unit testing requirements where this doesn't work?